### PR TITLE
[13.0][ADD] stock_quant_barcode: New module

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT28846
ping @chienandalu @carlosdauden

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]

File: stock_quant_barcode/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2021 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Stock Quant Barcode",
+    "summary": "Identify stock quants by barcode",
+    "version": "13.0.1.0.0",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "website": "https://odoo-community.org/",
```
Comment (victoralmau) [inline stock_quant_barcode/__manifest__.py:8]:
```suggestion
    "website": "https://github.com/OCA/stock-logistics-barcode",
```

File: stock_quant_barcode/models/stock_quant.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            barcode = vals.pop("barcode", None)
+            self.sudo().with_context(barcode_write=True).write({"barcode": barcode})
+        if not vals:
+            # Avoid to call super if vals has not values due to access rules errors
+            return True
+        return super().write(vals)
+
+    @api.model
+    def create(self, vals):
+        if "force_barcode" in self.env.context:
```
Comment (victoralmau) [inline stock_quant_barcode/models/stock_quant.py:22]:
```suggestion
        if self.env.context.get("force_barcode"):
```

File: stock_quant_barcode/models/stock_quant.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+from odoo import api, fields, models
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    barcode = fields.Char("Barcode", copy=False, index=True)
+
+    def write(self, vals):
+        if "barcode" in vals and "barcode_write" not in self.env.context:
```
Comment (victoralmau) [inline stock_quant_barcode/models/stock_quant.py:12]:
```suggestion
        if vals.get("barcode") and self.env.context.get("barcode_write"):
```

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Comment by sergio-teruel** [comment]
Included in #369

**Pull request closed without merging**